### DISCUSSION TITLE
Creating a date and adding days onto the date

### DISCUSSION THREAD

**Discussion by gruff101** [SilentVoid13/Templater#1121] [Help] [answered]
Hi,

I'm creating a week planning note and want to create the dates for each of the days based on the date in the filename.

I'm extracting the date which is in the title of the file. The regex format is [0-9]{4}).([0-9]{2}).([0-9]{2}
Extracted values are:
- YYYY
- MM
- DD

I thought I could:
1. Create new date using the values extracted using the "new Date" function 
2. Then add 1, 2, 3 etc. onto the date

This didn't work. Does anyone know how to do this?

**Comment by Zachatoo** [comment]
We can modify one of the [examples](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/date-module.html#examples) found in the docs to do what you want.

For example, if your filename is formatted as `YYYY-MM-DD`, then you can use the following, no regex needed.

```js
<% tp.date.now("YYYY-MM-DD", 1, tp.file.title, "YYYY-MM-DD") %>
<% tp.date.now("YYYY-MM-DD", 2, tp.file.title, "YYYY-MM-DD") %>
<% tp.date.now("YYYY-MM-DD", 3, tp.file.title, "YYYY-MM-DD") %>
<% tp.date.now("YYYY-MM-DD", 4, tp.file.title, "YYYY-MM-DD") %>
etc...
```

If your filename has other characters in there, you can still avoid regex by escaping characters in the format. For example, if your filename was `Weekly note 2023-05-17`, then you could do the following:

```js
<% tp.date.now("YYYY-MM-DD", 1, tp.file.title, "[Weekly note] YYYY-MM-DD") %>
```

Relevant docs explaining the four arguments we're passing to `tp.date.now` [here](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/date-module.html#tpdatenowformat-string--yyyy-mm-dd-offset-numberstring-reference-string-reference_format-string).

**Reply by Zachatoo** [reply, marked as answer]
Just realized you're using `YYYY.MM.DD` instead of `YYYY-MM-DD`, but should be able to swap out the reference format.

```js
<% tp.date.now("YYYY-MM-DD", 1, tp.file.title, "YYYY.MM.DD") %>
```

**Reply by gruff101** [reply]
As always thanks @Zachatoo. Managed to tweak the function you gave me. 

The part I was confused about was the function as I assumed "date.now" related to the date as of today. Learnt something new which is great!

Tested passing in a different date as a string which also worked. Thanks again.
